mantidimaging/gui/windows/image_load_dialog/presenter.py

```diff
@@ -47,126 +46,95 @@ def do_update_field(self, field: Field) -> None:
 
-        for file_info in FILE_TYPES:
-            if file_info.mode == "images":
-                field = self.view.fields[file_info.fname]
-                images = find_images(sample_dirname,
-                                     file_info.tname,
-                                     suffix=file_info.suffix,
-                                     look_without_suffix=file_info.suffix == "Before",
-                                     image_format=self.image_format)
-                field.set_images(images)
-            elif file_info.mode == "180":
-                field = self.view.fields[file_info.fname]
-                field.path = find_180deg_proj(sample_dirname, self.image_format)
-
-        try:
-            self.set_sample_log(self.view.fields["Sample Log"], self.last_file_info.filenames)
-        except RuntimeError as err:
-            self.view.show_error(str(err), traceback.format_exc())
-
-        self.view.fields["Sample Log"].use = False
-
-        for pos in ["Before", "After"]:
-            flat_path = self.view.fields[f"Flat {pos}"].path_text()
-            if flat_path:
-                log_path = find_log_for_image(Path(flat_path))
-                if log_path:
-                    self.view.fields[f"Flat {pos} Log"].path = log_path
-                    self.view.fields[f"Flat {pos} Log"].use = False
-
-        self.view.images_are_sinograms.setChecked(self.last_file_info.sinograms)
-
-        self.view.sample.update_indices(self.last_file_info.shape[0])
-        self.view.sample.update_shape(self.last_file_info.shape[1:])
+        sample_field.widget.setExpanded(True)
+        sample_shape = read_image_dimensions(Path(selected_file))
+        self.view.sample.update_indices(len(sample.all_indexes))
+        self.view.sample.update_shape(sample_shape)
         self.view.enable_preview_all_buttons()
         self.view.ok_button.setEnabled(True)
 
-    def do_update_flat_or_dark(self, field: Field, selected_file: Optional[str]) -> None:
-        suffix = field.file_info.suffix
-        if not selected_file:
-            return
-        selected_dir = Path(os.path.dirname(selected_file))
-        images = find_images(selected_dir, field.file_info.fname, suffix, image_format=self.image_format)
-        if not images:
-            base_name = os.path.basename(selected_file).rpartition("_")[0]
-            images = find_images(selected_dir, base_name, "", image_format=self.image_format)
-        field.set_images(images)
+        for file_info in FILE_TYPES:
+            if file_info.mode in ["images", "180"]:
+                related_group = sample.find_related(file_info)
+                if related_group:
+                    self.update_field_with_filegroup(file_info, related_group)
+
+    def update_field_with_filegroup(self, file_info: FILE_TYPES, file_group: FilenameGroup):
+        file_group.find_all_files()
+        file_list = list(file_group.all_files())
+        self.view.fields[file_info.fname].set_images(file_list)
```

Comment:
I think I'll leave this for now. A future change I've been pondering is having `Field` hold the `FilenameGroup`. This would save `get_parameters` from having to recreate the `FilenameGroup`.  #1733

Reply:
Placing `FileNameGroup` within `Field` does sound preferable and possibly cleaner if we would no longer need to recreate the `FileNameGroup` within `get_parameters`